pkg/watermark/fetch/edge_fetcher_set.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2023 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fetch
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/isb"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"github.com/numaproj/numaflow/pkg/watermark/wmb"
+	"go.uber.org/zap"
+)
+
+// a set of EdgeFetchers, incoming to a Vertex
+// (In the case of a Join Vertex, there are multiple incoming Edges)
+// key=name of From Vertex
+type edgeFetcherSet struct {
+	edgeFetchers map[string]Fetcher
+	log          *zap.SugaredLogger
+}
+
+func NewEdgeFetcherSet(ctx context.Context, edgeFetchers map[string]Fetcher) Fetcher {
+	return &edgeFetcherSet{
+		edgeFetchers,
+		logging.FromContext(ctx),
+	}
+}
+
+// ProcessOffsetGetWatermark processes the offset on the partition indicated and returns the overall Watermark
+// from all Partitions
+func (efs *edgeFetcherSet) ProcessOffsetGetWatermark(inputOffset isb.Offset, fromPartitionIdx int32) wmb.Watermark {
+	_ = efs.ProcessOffset(inputOffset, fromPartitionIdx) // even if it errored, we'll keep going
```

Comment:
I didn't log it here since it's being logged in the underlying function 
As far as why we keep going, I guess we don't need to, but I figured it was better to try if for some reason something happens. ?

Reply:
I just checked underlying code, looks like the only error will be from `var offset, err = inputOffset.Sequence()` and this is not a recoverable error...

we can discuss with Vigith and Derek on how to proceed